gateway/reverse_proxy.go

```diff
@@ -1146,3 +1179,26 @@ func nopCloseResponseBody(r *http.Response) {
 
 	copyResponse(r)
 }
+
+func IsUpgrade(req *http.Request) (bool, string) {
+	if !config.Global().HttpServerOptions.EnableWebSockets {
+		return false, ""
+	}
+
+	contentType := strings.ToLower(strings.TrimSpace(req.Header.Get(headers.Accept)))
+	if contentType == "text/event-stream" {
+		return true, ""
+	}
+
+	connection := strings.ToLower(strings.TrimSpace(req.Header.Get(headers.Connection)))
+	if connection != "upgrade" {
+		return false, ""
+	}
+
+	upgrade := strings.ToLower(strings.TrimSpace(req.Header.Get("Upgrade")))
+	if upgrade != "" {
```

Comment:
Upgrade should be checked for correct upgrade I think:
https://en.wikipedia.org/wiki/HTTP/1.1_Upgrade_header
The original function did this correctly.

Reply:
On of the reasons I do not check it now, because it is more general-purpose, and not limited to websockets. Original reverse proxy Go implementation does the same logic, and just checks if field non empty.